File: CPP/110A.cpp
#include <iostream>

using namespace std;

int main() {
  string xs;
  cin >> xs;

  size_t count = 0;
  for(const char& x: xs) 
    if(x == '4' || x == '7') count++;

  if(count == 4 || count == 7)
    cout << "YES" << endl;
  else
    cout << "NO" << endl;
}

File: Haskell/110A.hs
main :: IO ()
main = interact solve

solve :: String -> String
solve s = if count == 4 || count == 7 then "YES" else "NO"
  where 
    count = length $ filter (\x -> x == '7' || x == '4') s

-- main :: IO ()
-- main = interact $ solve .  read

-- count :: Int -> Int
-- count x 
--   | r == 4 || r == 7 = 1 + count q
--   | q == 0 = 0
--   | otherwise = count q
--   where 
--     (q, r) = x `divMod` 10

-- solve :: Int -> String
-- solve x = if count' == 4 || count' == 7 then "YES" else "NO"
--   where 
--     count' = count x